### DISCUSSION TITLE
Builder undefined error (dupe?)

### DISCUSSION THREAD

**Discussion by romelussaladar** [aclap-dev/video-downloadhelper#72] [Help and Q&A]
### Your question

Hello SUPPORT..

I am using CHROME and VIDEO DOWNLOAD HELPER Auto UPDATED to the latest 8.2.0.19

I was DOWNLOADING videos BEFORE THIS update working all FINE...

ONLY after the UPDATE occurred... the NEXT VIDEO ATTEMPT DOWNLOAD kept showing the folliwng MESSAGE ERROR

**No download strategy for builder: undefined
Error: No download strategy for builder: undefined
    at yT (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:44:6803)
    at async Zr (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:44:12666**

I attempted SEVERAL TIMES the SAME VIDEO to download... and THE ABOVE ERROR kept adding up SEVERAL TIMES.

ANY ideas ???

**Comment by paulrouget** [comment]
Please don't use caps.

What's the website?

**Reply by romelussaladar** [reply]
Oh it;s not a URL... I was on TWITCH... and it gives a FEATURE to CLIP a LIVE STREAM or a PART of a VIDEO being streamed... and that CLIPPING action CREATES a NEW PAGE or TAB with a 1min 30 second SEGMENT... and in THAT NEW PAGE we can extend a PORTION of the CLIPPING from 30 seconds UP TO 1 MINUTE of the 1min 30 second SEGMENT. I use VIDEODOWNLOADER to GRAB that FULL 1min 30second SEGMENT that happens to be in the MEMORY that VIDEODOWNLOADER can GRAB to DOWNKOAD, and THAT NEW PAGE has NO DEFINED WEBPAGE.. because after CLIPPING a PORTION of that FULL SEGMENT... then it CREATES a DEFINED URL for that CLIP PORTION...

HOPE that makes sense...

PRIOR to the UPDATE... I could ALWAYS TAKE that FULL 1minute 30 second CLIP SEGMENT and DOWNLOAD it FULL with VIDEODOWNLOADER...

Because of THE CURRENT ERRORS... I still have THE NEW PAGE still OPEN on my CHROME BROWSER, but I CURRENTLY cannot use VIDEO DOWNLOAD HELPER to grab the VIDEO... ALL ERROR OUTPUT MESSAGE as mentioned ABOVE...

Oh sorry for the caps... just making emphasis..

**Reply by paulrouget** [reply]
> HOPE that makes sense...
> Oh sorry for the caps... just making emphasis..

Sorry, this is impossible to read. Can you please rewrite this in a proper form?

And please share the Twitch url.

**Reply by romelussaladar** [reply]
Oh...  well the twitch clip page is created when we click on the CLIP icon in the stream...

For example if this is the TWITCH STREAMER currently active now... https://www.twitch.tv/hanavocado

and when I click the CLIP button on the STREAM...  it creates this new page showing the clip to adjust https://clips.twitch.tv/create

then inside this new page, we adjust the size of the clip to publish.

Usually in the past when I use Videodownload helper, I can grab this whole 1 minute 30 second segment. I can't do it now after the new update.

**Reply by paulrouget** [reply]
Please try again tomorrow once 8.2.0.20 is available.

**Comment by paulrouget** [comment]
> No download strategy for builder: undefined

I think I found the bugs. I updated the addon. Tomorrow you'll see version `8.2.0.20` which hopefully fixes your problem.

**Reply by romelussaladar** [reply]
Hello Paul,

The latest AUTOupdate on my CHROME now shows version 8.2.0.21 . This update now appears to download from the twitch clips page, HOWEVER it changes the previous version of the default downloaded directory. Also I noticed the downloaded speed seems to be significantly slower than before. I tested downloading other items in skype and telegram, and those speeds were quite fast and appeared immediate, so it doesn't appear to be the internet speeds. The reason why I noticed, is because I still kept the open CLIP pages from twitch, without closing them for the last 2 days, waiting to use the current auto-update of videodownload helper.

Just letting you be aware that the current update does work, but it seems to redirect to different directories (not the original download folder) which is where all the other twitch clips were always downloaded to. This time it defaulted to dwhelper directory, which seemed to ONLY be the folder for livestream recorded files.

Hope this info helps.

Will also see what happens with the next update as you suggested in the discussion forums.

Cheers,

Great work !!!

**Reply by romelussaladar** [reply]
Hello @paulrouget  ... in addition to my last reply, when I downloaded using the 8.2.0.21 version, the files are not downloaded with MP$ automatically put on the filename. I had to download all the clips adding mp4 as the file type. I only noticed this the first 3 times I downloaded the clips, which I had to change later within windows. All subsequent clips being downloaded, I just added *.mp4 as the full filename and file type being downloaded.

Hope this helps with current observations.

Cheers mate

**Reply by GrampaWildWilly** [reply]
If you are used to having Download or Quick Download as your default action, it is possible that your default action has changed.  I saw this on an update within the past week or so.  You just need to set your default action again & it should be fine from that point on.  As for the target of your download, you also have to do at least one Download & look closely at where it's downloading to.  That's a Download, not a Quick Download.  Be sure you do not do a Quick Download for this.  Once you do one Download into your desired directory, subsequent downloads, Quick or not, should pick up that directory automatically.

**Reply by romelussaladar** [reply]
Hi @GrampaWildWilly ,

Just read your reply... I've never used the Quick Download, and for the TwitchClips in the past it has always downloaded into the windows download folder. When I am grabbing an actual Twitch LIVESTREAM, then for the time I grab the video until I stop that livestream grab, then it in the past has always saved auto to the DWHELPER folder. For the recent update of 8.2.0.21 it has just defaulted now to the Dwhelper folder, and it doesn't save the file as ***mp4***, so I always have to type the extension for all the videos for this evening AUS evening time.

These are just current observations for @paulrouget to be aware of. Also when I am grabbing downloading the Twitch Livestream, it doesn't show the total size being downloaded, only showing the download bit rate being grabbed live. Previously I could watch the size of the livestream, as it increases the longer I leave it. Current update doesn't do this anymore.

**Reply by paulrouget** [reply]
> it doesn't save the file as mp4

What's the extension then?

**Reply by romelussaladar** [reply]
Oh for me, it had nothing... at first I saved the first 3 files and it didn't have any extension that VLC would identify as a VIDEO file. I had to go into advanced settings to rename the file as an mp4.  Afterwards I just made sure to add the mp4 extension to all the other clips that I am also now still downloading.

**Reply by romelussaladar** [reply]
***CORRECTION to the last REPLY***

I am now using 8,2.0.23 VERSION and now it shows the mp4 extension.

**Reply by romelussaladar** [reply]
***NEW observed issue with FILE saving***

I am currently using 8.2.0.23, and I noticed NOW with every new download, when saving the file into the same directory, I do not see the previous file name that was recently downloaded at the top of the list, even when I change settings to show the date latest file download etc. I have to go out into the windows folder to see the files downloaded and copy the file names to be the additional parts to the previous videos downloaded. In the past with working versions of VIDEOdownload helper, I could always see the previous file, and then select that file so it appears in the filename to be saved for the current video about to be downloaded, then I can just put a number at the end for ****2 etc for each next part downloaded file. NOW there are no previous files that I can see that were downloaded before by VideoDownloadhelper.

All previous versions I could see previously downloaded files, including other files in the same directory. For some reason, no files downloaded can be seen when saving the current video about to be saved. This has been observed during the last 40++ files already downloaded.

**Reply by paulrouget** [reply]
Can you open another post for this? It's not about undefined builder anymore.

And also, please describe a little more your issue. I don't really understand what you mean.

Thank you.
